File: archive/canvas-api/src/test/java/MasteryPathUnitTest.java
/*
 * Copyright (C) 2017 - present Instructure, Inc.
 *
 *     Licensed under the Apache License, Version 2.0 (the "License");
 *     you may not use this file except in compliance with the License.
 *     You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 *     Unless required by applicable law or agreed to in writing, software
 *     distributed under the License is distributed on an "AS IS" BASIS,
 *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *     See the License for the specific language governing permissions and
 *     limitations under the License.
 *
 */

import com.google.gson.Gson;
import com.instructure.canvasapi.model.MasteryPath;
import com.instructure.canvasapi.utilities.CanvasRestAdapter;

import junit.framework.Assert;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.annotation.Config;



@Config(sdk = 17)
@RunWith(RobolectricGradleTestRunner.class)
public class MasteryPathUnitTest extends Assert {

    @Test
    public void testMasteryPath(){
        Gson gson = CanvasRestAdapter.getGSONParser();
        MasteryPath masteryPath= gson.fromJson(masteryPathsJSON, MasteryPath.class);

        assertNotNull(masteryPath);
        assertFalse(masteryPath.isLocked());
        assertNotNull(masteryPath.getAssignmentSets());

    }

    private String masteryPathsJSON =
                    " {\n" +
            "       \"locked\": false,\n" +
            "      \"assignment_sets\": [\n" +
            "        {\n" +
            "          \"id\": 2,\n" +
            "          \"scoring_range_id\": 2,\n" +
            "          \"created_at\": \"2016-08-03T19:04:44.860Z\",\n" +
            "          \"updated_at\": \"2016-08-03T19:04:44.860Z\",\n" +
            "          \"position\": 1,\n" +
            "          \"assignments\": [\n" +
            "            {\n" +
            "              \"id\": 2,\n" +
            "              \"assignment_id\": \"5\",\n" +
            "              \"created_at\": \"2016-08-03T19:04:44.865Z\",\n" +
            "              \"updated_at\": \"2016-08-03T19:04:44.865Z\",\n" +
            "              \"override_id\": 8,\n" +
            "              \"assignment_set_id\": 2,\n" +
            "              \"position\": 1,\n" +
            "              \"model\": {\n" +
            "                \"id\": 5,\n" +
            "                \"title\": \"Quiz 1~1\",\n" +
            "                \"description\": \"\",\n" +
            "                \"due_at\": null,\n" +
            "                \"unlock_at\": null,\n" +
            "                \"lock_at\": null,\n" +
            "                \"points_possible\": 0,\n" +
            "                \"min_score\": null,\n" +
            "                \"max_score\": null,\n" +
            "                \"grading_type\": \"points\",\n" +
            "                \"submission_types\": [\n" +
            "                   \"online_quiz\"],\n" +
            "                \"workflow_state\": \"published\",\n" +
            "                \"context_id\": 1,\n" +
            "                \"context_type\": \"Course\",\n" +
            "                \"updated_at\": \"2016-08-11T17:32:34Z\",\n" +
            "                \"context_code\": \"course_1\"\n" +
            "              }\n" +
            "            }\n" +
            "          ]\n" +
                    "}\n" +
                    "]\n" +
            " }";
}

File: libs/canvas-api-2/src/test/java/com/instructure/canvasapi2/unit/MasteryPathUnitTest.kt
/*
 * Copyright (C) 2017 - present Instructure, Inc.
 *
 *     Licensed under the Apache License, Version 2.0 (the "License");
 *     you may not use this file except in compliance with the License.
 *     You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 *     Unless required by applicable law or agreed to in writing, software
 *     distributed under the License is distributed on an "AS IS" BASIS,
 *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *     See the License for the specific language governing permissions and
 *     limitations under the License.
 *
 */

package com.instructure.canvasapi2.unit

import com.instructure.canvasapi2.models.MasteryPath
import com.instructure.canvasapi2.utils.parse
import org.intellij.lang.annotations.Language
import org.junit.Assert
import org.junit.Test

class MasteryPathUnitTest : Assert() {

    @Test
    fun testMasteryPath() {
        val masteryPath: MasteryPath = masteryPathsJSON.parse()

        Assert.assertNotNull(masteryPath)
        Assert.assertFalse(masteryPath.isLocked)
        Assert.assertNotNull(masteryPath.assignmentSets)

    }

    @Language("JSON")
    private val masteryPathsJSON = """
      {
        "locked": false,
        "assignment_sets": [
          {
            "id": 2,
            "scoring_range_id": 2,
            "created_at": "2016-08-03T19:04:44.860Z",
            "updated_at": "2016-08-03T19:04:44.860Z",
            "position": 1,
            "assignments": [
              {
                "id": 2,
                "assignment_id": "5",
                "created_at": "2016-08-03T19:04:44.865Z",
                "updated_at": "2016-08-03T19:04:44.865Z",
                "override_id": 8,
                "assignment_set_id": 2,
                "position": 1,
                "model": {
                  "id": 5,
                  "title": "Quiz 1~1",
                  "description": "",
                  "due_at": null,
                  "unlock_at": null,
                  "lock_at": null,
                  "points_possible": 0,
                  "min_score": null,
                  "max_score": null,
                  "grading_type": "points",
                  "submission_types": [
                    "online_quiz"
                  ],
                  "workflow_state": "published",
                  "context_id": 1,
                  "context_type": "Course",
                  "updated_at": "2016-08-11T17:32:34Z",
                  "context_code": "course_1"
                }
              }
            ]
          }
        ]
      }"""
}
